website/docs/tutorials/detect-race-conditions/one-step-closer-to-real-usages.mdx

```diff
@@ -35,16 +35,12 @@ pendingQueries.push(queued(2).then((v) => seenAnswers.push(v)));
 await s.waitFor(Promise.all(pendingQueries));
 ```
 
-An alternative options would have been to use `waitAll` instead of `waitFor` but it comes with a precise requirements being that promise had to be imperatively scehduled by the time we reach it. In other words, if our code delays a little bit the call to `call` it will not wait anything. With the current implementation of `queue` it works, but with next versions of it it might not be the case anymore as queued calls might start queuing themselves behind already started and running promises.
+An alternative would have been to use `waitAll` instead of `waitFor` but it comes with a precise requirement: promises have to be imperatively scheduled by the time we reach them. In other words, if our code delays a little bit the call to `call`, the scheduler might not wait enough.
```

Comment:
```suggestion
An alternative would have been to use `waitAll` instead of `waitFor` but it comes with a precise requirement: promises have to be already scheduled by the time we request the scheduler. In other words, if our code delays a little bit the call to `call`, the scheduler might not wait enough.
```